lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -3475,27 +3481,28 @@ swift::ModuleDecl *SwiftASTContext::GetModule(const FileSpec &module_spec,
   return NULL;
 }
 
-swift::ModuleDecl *
-SwiftASTContext::FindAndLoadModule(const SourceModule &module, Process &process,
-                                   Status &error) {
+template<typename ModuleT> swift::ModuleDecl *
+SwiftASTContext::FindAndLoadModule(const ModuleT &module, Process &process,
+                                   bool import_dylib, Status &error) {
   VALID_OR_RETURN(nullptr);
 
-  swift::ModuleDecl *swift_module = GetModule(module, error);
+  bool cached = false;
+  swift::ModuleDecl *swift_module = GetModule(module, error, &cached);
+  
   if (!swift_module)
     return nullptr;
-  LoadModule(swift_module, process, error);
-  return swift_module;
-}
 
-swift::ModuleDecl *
-SwiftASTContext::FindAndLoadModule(const FileSpec &module_spec,
-                                   Process &process, Status &error) {
-  VALID_OR_RETURN(nullptr);
+  if (import_dylib) {
```

Comment:
I'm missing something here.  It looks like the only time we call into this function with import_dylib == true is in CacheUserImports (i.e. when the user says `(lldb) expr import Foo`.   Everywhere else we pass in false.  If that's right the only place where GetSwiftAutoImportFrameworks is consulted is for User Imports?  But I thought we always had to import backing dylibs for User Imports, so that seems odd.  And if I'm reading this correctly there's no way to force us to import backing dylibs for CU imports, which I thought was the point of the AutoImportFrameworks setting.

The `if (cached)` test also seems odd.  If the module was not cached, we are going to try to import it anyway.  But if it is cached, we're going to check the setting.  Not sure what that logic is for.  This at least deserves some comment explaining the algorithm, as it wasn't clear to me.

Reply:
Very good point! This made me find a bug in `CacheImports`, which didn't distinguish between implicit and explicit imports. Fixed now.